_pytest/cacheprovider.py

```diff
@@ -147,11 +142,11 @@ def pytest_collection_modifyitems(self, session, config, items):
                     previously_failed.append(item)
                 else:
                     previously_passed.append(item)
-            if not previously_failed and previously_passed:
```

Comment:
when we have failed tests that are outside of the of the selection thats currently being executed, that happens